src/Agent.Worker/Handlers/ProcessHandler.cs

```diff
@@ -126,6 +129,8 @@ public async Task RunAsync()
             {
                 // Format the command so the environment variables can be captured.
                 cmdExeArgs = $"/c \"{command} {arguments} && echo {OutputDelimiter} && set \"";
+
+                _readonlyEnvVariables = Constants.Variables.ReadOnlyVariables.ConvertAll(var => VarUtil.ConvertToEnvVariableFormat(var));
```

Comment:
One point missed here - we shoud also collect runtime env variables.
we may get full list of readonly variables from the `AZP_READONLY_ENV_VARIABLES` implemented here: https://github.com/microsoft/azure-pipelines-agent/pull/4175